src/libbson/src/bson/bson-context.c

```diff
@@ -264,79 +139,261 @@ _bson_context_get_hostname (char *out)
 }
 
 
-static void
-_bson_context_init_random (bson_context_t *context, bool init_sequence)
+/*** ========================================
+ * The below SipHash implementation is based on the original public-domain
+ * reference implementation from Jean-Philippe Aumasson and DJB
+ * (https://github.com/veorq/SipHash).
+ */
+
+/* in-place rotate a 64bit number */
+void
+_rotl64 (uint64_t *p, int nbits)
```

Comment:
```suggestion
_bson_rotl64 (uint64_t *p, int nbits)
```

The linked [patch build](https://spruce.mongodb.com/task/mongo_c_driver_windows_2017_debug_compile_sasl_openssl_cse_patch_7a35cf865629122267c3d59cca0f922c5e455f0d_61e9fd0ad1fe074fa7152a7c_22_01_21_00_23_40/logs?execution=0) errors from redefinition:

```
c:\data\mci\a7a6ccecd92f7a5293b119f04a16cb5c\mongoc\src\libbson\src\bson\bson-context.c(151): error C2169: '_rotl64': intrinsic function, cannot be defined [C:\data\mci\a7a6ccecd92f7a5293b119f04a16cb5c\mongoc\src\libbson\bson_static.vcxproj]
```

Reply:
Oops. That's a fun coincidence.